### PR TITLE
refactor: remove duplicate code from unit tests

### PR DESCRIPTION
**Pull request by Zimann**
**Test file changes**

- card-footer.component.spec.ts
Removed the deprecated async() and replaced it with waitForAsync()

- carousel.component.spec.ts
Extracted the duplicated logic and placed it inside a function expression which gets called in multiple places

- feature-toggle.guard.spec.ts 
Created a separate enum to check for guard type when testing and the duplicated logic was placed inside a function expression which gets tweaked with different parameters that cater to the different test cases

**Style file changes**

-accordion-panel-heading.component.scss 
A simple class nesting change 

-button.component.scss 
Used the short-hand version of : top,left,right,bottom: 0 -> inset: 0 

**Directive file changes**

-feature-toggle.directive.ts 
Used the optional chaining operator and removed the check for the subscription

**Html file changes**

- modal-body-timer.component.html
'this' is not needed when referencing public component properties

**TS file changes**
- brand-icon.component.ts 
Removed the variable  _colour: string; as this was not used anywhere

## Requirements

Testing if all the unit tests pass like before with the command: npm run test 
Visually testing the components affected by the style changes (button and brand-icon components)

Storybook link: (once netlify has deployed link provide a link to the component)
Design link: (link to design or delete if not required)
Screenshot: (if appropriate provide a quick screen grab of the changes)

# Checklist:

- [ ] The commit messages follow the [convention for this project](./blob/master/docs/CONTRIBUTING.md#conventional-commits)
- [ ] I have provided an adequate amount of test coverage
- [ ] I have added the functionality to the [test app](./blob/master/docs/CONTRIBUTING.md#build-test-application)
- [ ] I have provided a story in storybook to document the changes
- [ ] I have provided documentation in the notes section of the story
- [ ] I have added any new public feature modules to public-api.ts

### CODE REVIEW

File: projects/canopy/src/lib/accordion/accordion-panel-heading/accordion-panel-heading.component.scss [hunk 1]
```diff
@@ -41,7 +41,9 @@
     pointer-events: none;
   }
 
-  &--active .lg-accordion__heading-icon {
-    transform: rotateX(180deg);
+  &--active {
```
Comment (elenagarrone) [inline projects/canopy/src/lib/accordion/accordion-panel-heading/accordion-panel-heading.component.scss:44]: I think it's ok to have this on a single line instead of nested since the `&--active` is not used by anything else 🤔

File: projects/canopy/src/lib/accordion/accordion-panel-heading/accordion-panel-heading.component.scss [hunk 2]
```diff
@@ -41,7 +41,9 @@
...
+    .lg-accordion__heading-icon {
```
Comment (andy-polhill) [inline projects/canopy/src/lib/accordion/accordion-panel-heading/accordion-panel-heading.component.scss:45]: @Zimann this was the only one I didn't quite get, although I think it may be a lack of understanding on my side, are these the same thing?

File: projects/canopy/src/lib/button/button.component.scss [hunk 3]
```diff
@@ -40,10 +40,7 @@
 
   &__spinner {
     position: absolute;
-    left: 0;
-    right: 0;
-    top: 0;
-    bottom: 0;
+    inset: 0;
```
Comment (elenagarrone) [inline projects/canopy/src/lib/button/button.component.scss:43]: Nice, I didn't know this property existed x) 
The only thing is that it's not supported by IE11 which I don't think should be an issue since we are not supporting it in our apps anymore, @andy-polhill have we decided if IE11 support is dropped for Canopy too?

File: projects/canopy/src/lib/accordion/accordion-panel-heading/accordion-panel-heading.component.scss (hunk 2 above)
Comment (Zimann) [reply projects/canopy/src/lib/accordion/accordion-panel-heading/accordion-panel-heading.component.scss:45]: Yep, it's the same thing. It's just making use of the nested selectors from scss instead of having it chained (.classA .classB -> can be written as : .classA { .classB }

File: projects/canopy/src/lib/accordion/accordion-panel-heading/accordion-panel-heading.component.scss (hunk 1 above)
Comment (Zimann) [reply projects/canopy/src/lib/accordion/accordion-panel-heading/accordion-panel-heading.component.scss]: Reverted it to how it was.

File: projects/canopy/src/lib/button/button.component.scss (hunk 3 above)
Comment (owensgit) [reply projects/canopy/src/lib/button/button.component.scss:43]: We don't support in our team either, I would suggest we drop it for Canopy if we haven't already

File: projects/canopy/src/lib/feature-toggle/feature-toggle.directive.ts [hunk 4]
```diff
@@ -61,8 +61,6 @@ export class LgFeatureToggleDirective implements OnInit, OnDestroy {
 
   ngOnDestroy() {
     /* istanbul ignore else */
```
Comment (owensgit) [inline projects/canopy/src/lib/feature-toggle/feature-toggle.directive.ts:63]: Just wondering, do we need this line anymore?

**Review by owensgit — COMMENTED** [review]
Great work, and congrats on your first Canopy contribution! Just a couple of minor comments/questions from me.

**Comment by owensgit** [comment]
Also just wondering about commit messages. We tend to always include a scope, although I know that's optional in the conventional commit format. Do we wan to continue that trend here, or shall we keep it optional? One suggestion on the two commits here could be:

> refactor(unit-tests): remove duplicate code
> revert(accordion): nest accordion heading selector

File: projects/canopy/src/lib/accordion/accordion-panel-heading/accordion-panel-heading.component.scss (hunk 1 above)
Comment (owensgit) [reply projects/canopy/src/lib/accordion/accordion-panel-heading/accordion-panel-heading.component.scss]: I think this change belongs in the previous commit: `revert: selection on accordion-panel-heading.scss`

File: projects/canopy/src/lib/button/button.component.scss (hunk 3 above)
Comment (owensgit) [inline projects/canopy/src/lib/button/button.component.scss:43]: I think this SCSS change could be in it's own commit, as it's not part of the unit tests?

File: projects/canopy/src/lib/modal/modal-body-timer/modal-body-timer.component.html [hunk 5]
```diff
@@ -1 +1 @@
-{{ this.timer }}
+{{ timer }}
```
Comment (owensgit) [inline projects/canopy/src/lib/modal/modal-body-timer/modal-body-timer.component.html:1]: Also wondering if this should be a separate commit, as it doesn't look related to the unit test changes.

File: projects/canopy/src/lib/button/button.component.scss (hunk 3 above)
Comment (Zimann) [reply projects/canopy/src/lib/button/button.component.scss:43]: Sure, I will tidy up the commits and make them more granular. Thanks! :)

File: projects/canopy/src/lib/feature-toggle/feature-toggle.directive.ts (hunk 4 above)
Comment (Zimann) [reply projects/canopy/src/lib/feature-toggle/feature-toggle.directive.ts]: Removed this comment.

File: projects/canopy/src/lib/modal/modal-body-timer/modal-body-timer.component.html (hunk 5 above)
Comment (Zimann) [reply projects/canopy/src/lib/modal/modal-body-timer/modal-body-timer.component.html:1]: Placed this in a separate commit. See: 7e17a3ab16d28a24086f4d1e770a7ac65c7e097b

**Comment by andy-polhill** [comment]
Hey @Zimann thanks for updating and apologies for being so fussy about commit messages, but scope wise we normally just use the higher level component name and not the file itself...
```
refactor(card): remove deprecated async
refactor(feature-toggle): use optional chaining
refactor(modal): remove unneeded 'this' reference
```

**Review by elenagarrone — APPROVED** [review]
Thanks @Zimann 🎊

File: projects/canopy/src/lib/button/button.component.scss (hunk 3 above)
Comment (andy-polhill) [reply projects/canopy/src/lib/button/button.component.scss:43]: good question, I think there may be a couple of outliers, but I'm happy to start moving away from IE11 too. We should document or raise an issue somewhere about IE11 support.

**Review by andy-polhill — APPROVED** [review]
